lib/Drush/Sql/SqlBase.php

```diff
@@ -257,7 +257,10 @@ public function drop_or_create() {
       $this->drop($this->listTables());
     }
     else {
+      $original_spec = $this->db_spec;
       $this->createdb();
+      // Restore the original spec as create db and su() can modify the spec.
+      $this->db_spec = $original_spec;
```

Comment:
I didnt investigate, but it is a bit odd to wrap createdb here. Can this dance not be done inside createdb()?